src/docfx/build/schema/BuildSchemaDocument.cs

```diff
@@ -3,16 +3,97 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
 using System.Linq;
-using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Docs.Build
 {
     internal static class BuildSchemaDocument
     {
-        public static Task<(IEnumerable<Error> errors, PageModel result, DependencyMap dependencies)> Build()
+        private static readonly Type[] s_schemaTypes = new[] { typeof(LandingData) };
+        private static readonly IReadOnlyDictionary<string, Type> s_schemas = s_schemaTypes.ToDictionary(type => type.Name);
+
+        public static (IEnumerable<Error> errors, PageModel result, DependencyMap dependencies) Build(
+            Document file,
+            TableOfContentsMap tocMap,
+            ContributionInfo contribution)
         {
-            return Task.FromResult((Enumerable.Empty<Error>(), (PageModel)null, DependencyMap.Empty));
+            Debug.Assert(file.ContentType == ContentType.SchemaDocument);
+
+            var (errors, token, schema) = Parse(file);
+
+            if (!s_schemas.TryGetValue(schema, out var schemaType))
+            {
+                throw Errors.SchemaNotFound(schema).ToException();
+            }
+
+            var content = token.ToObject(schemaType);
+
+            // TODO: consolidate this with BuildMarkdown
+            var locale = file.Docset.Config.Locale;
+            var metadata = JsonUtility.Merge(Metadata.GetFromConfig(file), token.Value<JObject>("metadata"));
+            var (id, versionIndependentId) = file.Docset.Redirections.TryGetDocumentId(file, out var docId) ? docId : file.Id;
+
+            // TODO: add check before to avoid case failure
+            var (repoErrors, author, contributors, updatedAt) = contribution.GetContributorInfo(
+                file,
+                metadata.Value<string>("author"),
+                metadata.Value<DateTime?>("update_date"));
+
+            var title = metadata.Value<string>("title");
+
+            var model = new PageModel
+            {
+                PageType = schema,
```

Comment:
>PageType [](start = 16, length = 8)

is pageType == Reference or the schema? seems Reference is consistent with Conceptual? or we want to provide anther type?

Reply:
We could unify sdp with markdown, by then, PageType will be LandingData, Tutorial, Conceptual, etc.
Content will be in different shape depending on PageType


---
In reply to: [203635437](https://github.com/dotnet/docfx/pull/3098#discussion_r203635437) [](ancestors = 203635437)